### ISSUE TITLE
Current bazel_build.py -fdebug-prefix-map doesn't work with Bazel sandboxing in 0.15.0-homebrew

### ISSUE THREAD

**Issue by nglevin** [bazelbuild/tulsi#52]
This is specifically for the direct debug prefix map solution that exists in Tulsi open source, today. Currently, it doesn't understand how to obtain paths generated by the new Bazel sandbox for Darwin, so breakpoint level debugging doesn't work for open source Tulsi.

Confirmed the problem with the iOS sample app built via https://github.com/nglevin/bazel/blob/fe4936031e38dd33687261334b2225224e96c430/site/docs/tutorial/ios-app.md with Bazel 0.15.0-homebrew.

I believe this would be fixed by porting the "normalized path" debug prefix map solution to open source Tulsi and bazel_build.py, which requires a wrapped_clang update.

I filed a follow up for Bazel, which needs a specific feature flag ported to open source for the normalized, dotted path debug-prefix-map solution to work as well; https://github.com/bazelbuild/bazel/issues/5545

**Comment by DavidGoldman** [comment]
Thanks for reporting, I'll look into bazelbuild/bazel#5545 and see if we can update the CROSSTOOL and wrapped_clang to add support.

As a temporary fix I think disabling the sandboxing should work (see #15), adding `--strategy= ObjcCompile =standalone` should work.

**Comment by nglevin** [comment]
Thinking further on the impact of the Bazel sandboxing change, it seems that anything that referenced `bazel info execution_root` or `BAZEL_EXECUTION_ROOT` would be affected.


From what I saw, the sandboxing changes the built paths such that they're now of this form:
```
camael:tutorial nglevin$ nm -pa /Users/nglevin/Library/Developer/Xcode/DerivedData/iOSTestApp-bjjzfccpngjxdxeooeawrukisskp/Build/Products/Debug-iphonesimulator/ios-app.app/ios-app 
0000000000000000 - 00 0000    SO /private/var/tmp/_bazel_nglevin/03ffb3c21f1df3470cbb58b4f44c041a/sandbox/darwin-sandbox/3/execroot/__main__/ios-app/UrlGet/
0000000000000000 - 00 0000    SO AppDelegate.m
0000000000000000 - 03 0001   OSO /private/var/tmp/_bazel_nglevin/03ffb3c21f1df3470cbb58b4f44c041a/sandbox/darwin-sandbox/5/execroot/__main__/bazel-out/ios-x86_64-min11.0-applebin_ios-ios_x86_64-dbg/bin/ios-app/libUrlGetClasses.a(AppDelegate_5a163255d0469b984755379b430eb0c0.o)
```

The important change is the use of paths like `/private/var/tmp/_bazel_nglevin/03ffb3c21f1df3470cbb58b4f44c041a/sandbox/darwin-sandbox/3/execroot` and `/private/var/tmp/_bazel_nglevin/03ffb3c21f1df3470cbb58b4f44c041a/sandbox/darwin-sandbox/5/execroot`.

That will probably cause some other breakages in bazel_build.py's expectations with file caching and the use of the execution root for redirection.  Bummer. 😒


`bazel info execution_root` isn't so helpful, and `bazel info` information doesn't leave many hints of what to expect from sandboxed paths:

```
camael:tutorial nglevin$ bazel info execution_root
/private/var/tmp/_bazel_nglevin/03ffb3c21f1df3470cbb58b4f44c041a/execroot/__main__
```

I'll move on, don't worry, just wanted to share my findings here because I've never seen sandboxing in action before.

**Comment by DavidGoldman** [comment]
This should now be fixed as of cb9a6cd and Bazel 0.17.1.
